### PR TITLE
perf(storage-proofs-post): increase parallelism in fallback-post

### PR DESCRIPTION
**Pull request by dignifiedquire**
This improves the parallelism during both vanilla `prove` and vanilla `verify`.

### CODE REVIEW

**Review by cryptonemo — APPROVED** [review]
+1 for the enum removal

**Comment by dignifiedquire** [comment]
thanks to @robyhugeman666 for reminding me that we can do more here

**Comment by robyhugeman666** [comment]
Very honored to be involved.


------------------------------------------------------------------

File: storage-proofs/post/src/fallback/vanilla.rs [hunk 1]
```diff
@@ -507,82 +525,86 @@ impl<'a, Tree: 'a + MerkleTreeTrait> ProofScheme<'a> for FallbackPoSt<'a, Tree>
...
+                        .collect::<Result<Vec<bool>>>()?;
+
+                    Ok(is_valid_list.into_iter().all(|v| v))
+                })
+                .reduce(
+                    || Ok(true),
+                    |all_valid, is_valid| Ok(all_valid? && is_valid?),
+                )?;
+            if !is_valid {
+                return Ok(false);
```
Comment (free1139) [inline storage-proofs/post/src/fallback/vanilla.rs:607]: @dignifiedquire we also need continue the check and record the fault sectors here. I had lost the storage power cause some sectors failed in here, but i forgot the detail.

**Pull request merged**